api/src/main/java/org/openmrs/api/CohortService.java

```diff
@@ -193,5 +208,44 @@
 	 */
 	@Authorized( { PrivilegeConstants.GET_PATIENT_COHORTS })
 	public Cohort getCohortByUuid(String uuid);
+
+	/**
+	 * Adds membership to a Cohort
+	 *
+	 * @param cohort
+	 * @param cohortMembership
+	 * @return the Cohort that was passed in with the CohortMembership
+	 * @should add Membership to Cohort and save the Cohort
+	 */
+	@Authorized( {PrivilegeConstants.EDIT_COHORTS })
+	Cohort addMembershipToCohort(Cohort cohort, CohortMembership cohortMembership);
+
+	/**
+	 * Removes membership details to a Cohort
+	 *
+	 * @param cohort
+	 * @param cohortMembership
+	 * @return the Cohort that was passed in with the CohortMembership
+	 * @should removes Membership from Cohort by setting end date and save the Cohort
+	 */
+	@Authorized( {PrivilegeConstants.EDIT_COHORTS })
+	Cohort removeMemberShipFromCohort(Cohort cohort, CohortMembership cohortMembership);
+
+	/**
+	 * Void membership of Cohort for voided Patients
+	 *
+	 * @param patient
+	 * @should void the membership for the patient that was passed in
+	 */
+	@Authorized( {PrivilegeConstants.EDIT_COHORTS })
+	void patientVoided(Patient patient);
 	
+	/**
+	 * Unvoid membership of Cohort for unvoided Patients
+	 *
+	 * @param patient
```

Comment:
I think the former. Only because the purpose of these methods are for the event when a patient is voided or unvoided. If there is a better name, I am open to suggestions. Then again, I merely followed the [design forum notes](https://notes.openmrs.org/2016-10-12-Design-Forum).